src/cobovault.js

```diff
@@ -0,0 +1,313 @@
+ * const interaction = new CoboVaultExportExtendedPublicKey();
+ * const encodedString = readCoboVaultQRCode(); // application dependent
+ * const {xpub, bip32Path} = interaction.parse(encoodedString);
+ * console.log(xpub);
+ * // "xpub..."
+ * console.log(bip32Path);
+ * // "m/45'/0'/0'"
+ */
+
+export class CoboVaultExportExtendedPublicKey extends CoboVaultReader {
+    messages() {
+        const messages = super.messages();
+        messages.push(commandMessage({
+            instructions: "①Please open Cobo Vault, go to Menu > Multisig Wallet > More > Show/Export  XPUB. ②Click the camera icon above and scan the QR Code of XPUB displays on Cobo Vault",
+        }));
+        return messages;
+    }
+
+    parseResult(result) {
+        const {xpub, path: bip32Path, xfp} = result;
+        if (!xpub) {
+            throw new Error('No Extended PublicKey.');
+        }
+        if (!bip32Path) {
+            throw new Error("No BIP32 path.");
+        }
+        result.bip32Path = bip32Path;
+        result.xpub = xpub;
+        result.rootFingerprint = xfp;
+        Reflect.deleteProperty(result, "xfp");
+        return {
+            success: true,
+            result,
+            type: "json",
+        };
+    }
+
+    parse(workloads) {
+        try{
+            const {result} = super.parse(workloads);
+            return this.parseResult(result)
+        }catch (e) {
+            throw new Error("Unable to parse qr code")
+        }
+    }
+
+    parseFile(jsonString) {
+        try{
+            const result = JSON.parse(jsonString);
+            return this.parseResult(result)
+        }catch (e) {
+            throw new Error("Unable to parse file")
+        }
+    }
+}
+
+/**
+ * Returns signature request data to display in a QR code for CoboVault
+ * and reads the signature data passed back by CoboVault in another QR
+ * code.
+ *
+ * NOTE: Transactions with inputs & outputs to non-P2SH addresses are not supported by CoboVault.
+ *
+ * @extends {module:cobovault.CoboVaultDisplayer}
+ * @example
+ * const interaction = new CoboVaultSignMultisigTransaction({inputs, outputs, bip32Paths});
+ * console.log(interaction.request());
+ * // "IJQXGZI..."
+ *
+ * // Display a QR code containing the above data to CoboVault running
+ * // `sign-bitcoin` and it will return another QR code which needs
+ * // parsed.
+ * const encodedString = readCoboVaultQRCode(); // application dependent
+ * const signatures = interaction.parse(encoodedString);
+ * console.log(signatures);
+ * // ["ababa...01", ... ]
+ *
+ */
+export class CoboVaultSignMultisigTransaction extends CoboVaultReader {
+
+    /**
+     *
+     * @param {object} options - options argument
+     * @param {array<object>} options.inputs - inputs for the transaction
+     * @param {array<object>} options.outputs - outputs for the transaction
+     * @param {array<string>} options.bip32Paths - BIP32 paths
+     */
+    constructor({network, inputs, outputs, bip32Paths, psbt}) {
```

Comment:
if non-P2SH inputs or outputs are not supported by CoboVault - check it here and fail appropriately.